src/Vts/MonteCarlo/DataStructuresValidation/TissueInputs/BoundingCylinderTissueInputValidation.cs

```diff
@@ -47,23 +43,24 @@ public static ValidationResult ValidateInput(ITissueInput input)
             // check that layer definition is valid
             var tempResult = MultiLayerTissueInputValidation.ValidateLayers(layers);
 
-            if (!tempResult.IsValid){ return tempResult; }
+            if (!tempResult.IsValid) return tempResult;
 
             // test for air layers and eliminate from list
             var tissueLayers = layers.Where(layer => !layer.IsAir());
             
             // check that there is at least one layer of tissue 
-            if (!tissueLayers.Any())
+            var layerTissueRegions = tissueLayers.ToList();
+            if (!layerTissueRegions.Any())
             {
                 tempResult = new ValidationResult(
                     false,
                     "BoundingCylinderTissueInput: tissue layer is assumed to be at least a single layer with air layer above and below",
                     "BoundingCylinderTissueInput: redefine tissue definition to contain at least a single layer of tissue");
             }
 
-            if (!tempResult.IsValid) { return tempResult; }
+            if (!tempResult.IsValid) return tempResult;
 
-            var layersHeight = tissueLayers.Sum(layer => layer.ZRange.Delta);
+            var layersHeight = layerTissueRegions.Sum(layer => layer.ZRange.Delta);
 
```

Comment:
Did the functionality of this code change?

Reply:
This one was okay as is.